### DISCUSSION TITLE
To link a GitHub repository, you need to install the GitHub integration first.

### DISCUSSION THREAD

**Discussion by odiseo0** [vercel/vercel#7646] [Help] [answered]
Hello, I need help trying to figure out how could I build this object: https://vercel.com/docs/rest-api#:~:text=framework%20is%20selected-,gitRepository,-object

currently my request body looks like this:
```
{
  "gitRepository": {"branch":"master","repo":"https://github.com/username/repository.git", "repoId": repoid, "ref":"master","type":"github"},
  "name": "my-vercel-project"
}
```
When I do a post with that request body I get a 400 error with this message:
```
{'error': {'code': 'bad_request', 'message': 'To link a GitHub repository, you need to install the GitHub integration first.', 'action': 'Install GitHub App', 'link': 'https://github.com/apps/vercel', 'repo': ''}}
```
The thing is that I already have the GitHub application installed in my Vercel account, what could be causing the problem? or better
I'm building the `gitRepository` object  correctly? I haven't found any example of how it looks like.

**Comment by nzasepa** [comment, marked as answer]
The best way for you is to just have a look at the network requests within the Vercel dashboard. For example, this should be a valid `gitRepository` object:
```
{
  repo: 'username/repository'
  type: 'github',
  sourceless: true
}
```

**Reply by odiseo0** [reply]
Thanks, I'll check

**Reply by odiseo0** [reply]
This does the job well, thanks

**Comment by 0p3r4t0r** [comment]
Just in case anyone comes across this in the future. As stated in the accepted answer, be sure to specify your repo as `username/repository` and _not_ with the full path `https://www...`
